modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/utils/SystemCmd.java

```diff
@@ -83,12 +82,16 @@ public SystemCmdResult execute(String cmd, int waitSecond) {
             log.debug("stderr of the command: " + systemCmdResult.getCommandStdError());
         } catch (final IOException ex) {
             systemCmdResult.setCommandSuccessful(false);
+            systemCmdResult.setCommandStdError(ex.getMessage());
         } catch (final InterruptedException ex) {
             log.error("SystemCmd process be interrupted.");
             Thread.currentThread().interrupt();
             systemCmdResult.setCommandSuccessful(false);
+            systemCmdResult.setCommandStdError(ex.getMessage());
         } catch (ExecutionException e) {
-            throw new TerraformExecutorException(e.getMessage());
+            systemCmdResult.setCommandSuccessful(false);
+            systemCmdResult.setCommandStdError(e.getMessage());
+            throw new RuntimeException(e);
```

Comment:
We must throw TerraformExecutorException here and avoid throwing any java inbuilt exceptions.